### ISSUE TITLE
Accessibility Testing for Veteran Facing Forms - 21-0972 - Alternate Signer

### ISSUE THREAD

**Issue by tbaker1026** [department-of-veterans-affairs/va.gov-team#63255]
### Product information

- [X] Team name, product name, and feature name have been added to the title of this issue.
- [X] Team label, product label and feature label (if applicable) have been added to this issue.

### Who completed the use of color and color contrast test?

@jeana-adhoc 

### Use of color and color contrast checks

- [X] All text of 20px or smaller has a 4.5:1 contrast ratio to its background (or better)
- [X] All text of 20px or larger has a 3:1 contrast ratio to its background (or better)
- [X] Non-text elements have a 3:1  contrast ratio to their background and to neighboring elements (or better)
- [X] Color is not the only way to distinguish links from other text (eg. links are underlined)
- [X] Any charts, maps, infographics, and tables convey all information without only relying on color
- [X] Content does not refer to color, especially when providing user instructions (eg. "Click the blue button")

### How did color testing go?

No issues

### Who completed the axe scans?

@jeana-adhoc 

### axe checks

- [X] Each page has been scanned using axe (results shared below)
- [X] axe is integrated into your end-to-end testing

### axe DevTools scan results

| Axe Results  | Page |
| ------------- | ------------- |
| No issues  | [Introduction](https://staging.va.gov/supporting-forms-for-claims/alternate-signer-form-21-0972/introduction)  |
| No issues  | [Your personal information](https://staging.va.gov/supporting-forms-for-claims/alternate-signer-form-21-0972/preparer-personal-information)  |
| No issues     | [Your mailing address](https://staging.va.gov/supporting-forms-for-claims/alternate-signer-form-21-0972/preparer-address) |
| No issues | [Your contact information](https://staging.va.gov/supporting-forms-for-claims/alternate-signer-form-21-0972/preparer-contact-information) |
| No issues | [Who you'll be signing for](https://staging.va.gov/supporting-forms-for-claims/alternate-signer-form-21-0972/claimant-identification) |
| No issues | [Veteran's %%Variable%% information](https://staging.va.gov/supporting-forms-for-claims/alternate-signer-form-21-0972/claimant-personal-information) |
| No issues | [Veteran's %%Variable%% identification information](https://staging.va.gov/supporting-forms-for-claims/alternate-signer-form-21-0972/claimant-identification-information) |
| No issues | [Veteran's %%Variable%% mailing address](https://staging.va.gov/supporting-forms-for-claims/alternate-signer-form-21-0972/claimant-address) |
| No issues | [Veteran's %%Variable%% contact information](https://staging.va.gov/supporting-forms-for-claims/alternate-signer-form-21-0972/claimant-contact-information) |
| No issues  | [Qualifications 1](https://staging.va.gov/supporting-forms-for-claims/alternate-signer-form-21-0972/preparer-qualifications-1) |
| No issues | [Qualifications 2](https://staging.va.gov/supporting-forms-for-claims/alternate-signer-form-21-0972/preparer-qualifications-2) |
| No issues | [Veteran's personal information](https://staging.va.gov/supporting-forms-for-claims/alternate-signer-form-21-0972/veteran-personal-information) |
| No issues | [Veteran's Identification information 1](https://staging.va.gov/supporting-forms-for-claims/alternate-signer-form-21-0972/veteran-identification-information-1) |
| No issues | [Veteran's Identification information 2](https://staging.va.gov/supporting-forms-for-claims/alternate-signer-form-21-0972/veteran-identification-information-2) | 
| No issues | [Additional information](https://staging.va.gov/supporting-forms-for-claims/alternate-signer-form-21-0972/additional-information) | 
| No issues | [Review page](https://staging.va.gov/supporting-forms-for-claims/alternate-signer-form-21-0972/review-and-submit) |
| False positive Aria issue (Thanks to the shadowdom) | [Confirmation page](https://staging.va.gov/supporting-forms-for-claims/alternate-signer-form-21-0972/confirmation) |

### How did axe testing go?

Only one false-positive issue was reported by the axe tool on the "Expand all/Collapse all" button on the accordion. The same false positive also appears in storybook. If you take everything out of the shadowdom, there is no error.

<img width="450" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/6232d125-e084-43b3-a5b4-14202db0f600">

### Who completed the content zoom and reflow test?

@jeana-adhoc 

### Content zoom and reflow checks

- [x] All page elements are readable and usable at 200% zoom
- [ ] All page elements are readable and usable at 300% zoom
- [ ] All page elements are readable and usable at 400% zoom

### Who completed the keyboard navigation test?

@jeana-adhoc 

### Keyboard navigation checks

- [x] Each link, button, form input, checkbox, radio button, select menu, and custom element can receive keyboard focus
- [ ] Each link, button, form input, checkbox, radio button, select menu, and custom element responds to expected keys
- [x] All elements under focus have a visible focus indicator
- [x] The order of [Tab] stops made sense and was appropriate for completing tasks

### How did keyboard testing go?

Pretty good. Only one issue regarding accordions detailed below.

#### Radio groups on Windows machines   

[DST Ticket here](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2035)

I am unable to select radio buttons using the arrow keys in Windows Chrome or Edge while the NVDA or JAWS screen readers are running. I also cannot select radio buttons in the [v3 version of the radio group in Storybook](https://design.va.gov/storybook/?path=/docs/uswds-va-radio--default), however, the [v3 version of the radio group in USWDS](https://designsystem.digital.gov/components/radio-buttons/) is selectable by keyboard on a PC. I think this is going to be a DST bug, but I wanted to document it here.

#### Focus management   
[Ticket here](https://github.com/department-of-veterans-affairs/va.gov-team/issues/64220)

On the confirmation page there is a collection of accordions with an `expand / collapse` button. Currently, when you click that button (mouse or keyboard) the focus is lost, and the page scrolls. The focus should remain on the button and the page shouldn't scroll.  It does look like the focus is remaining on the button, but the page scrolls and shouldn't when the accordion opens. Here is a link to an [accordion in use with the desired behavior](https://staging.va.gov/manage-va-debt/#other-ways-to-make-a-payment). Below is a video of the undesired behavior.

https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/161f18a5-a4e3-4725-8174-744e5344bb2b

#### Accordion link list
The last 2 accordions don't format the links the same as the others. After I get clarification, I'll make a PR for this and fix it.

**Comment by jeana-adhoc** [comment]
Because of catch-all issues found on the form, it's best that we wait and finish this next sprint. I did very little work on it this sprint - enough to see that we should [wait until this ticket is taken care of](https://github.com/department-of-veterans-affairs/VA.gov-team-forms/issues/350#issuecomment-1680953532).

**Comment by tbaker1026** [comment]
@jeana-adhoc - want to work on this one before next Thursday staging review since UI/UX catch all is completed

**Comment by jeana-adhoc** [comment]
Testing is complete - there were 3 issues Identified:

1. Radio groups on Windows using NVDA or JAWS are un-navigable. [Ticket written for the DST](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2035) This impacts all of the v3 radio groups
2. Focus is lost/page scrolls when opening all accordions on the confirmation page. [A ticket was written for our forms team to work on](https://github.com/department-of-veterans-affairs/va.gov-team/issues/64220)
3. Content issues in the confirmation page. Still waiting guidance. Will update with a ticket/PR as soon as I can.
